config/initializers/default_url_options.rb

```diff
@@ -1,4 +1,9 @@
 uri = URI(HostingEnvironment.application_url)
-Rails.application.routes.default_url_options[:host] = uri.host
-Rails.application.routes.default_url_options[:port] = uri.port
-Rails.application.routes.default_url_options[:protocol] = uri.scheme
+application_config = Rails.application.routes
```

Comment:
I think this file's duties could be inlined in `config.application.rb`, which is a less surprising place to find this config. (though exactly how, via `config.*` or direct update, isn't clear — see https://github.com/rails/rails/issues/29992)

Reply:
I agree but setting these values in `Rails.application.routes` has so far been the most reliable way to configure url options in all contexts outside mailers and somehow `routes` does not seem to be available at the time `config/application.rb` is loaded. I'd rather keep all config in one place.